substrate/frame/support/src/traits/tokens/imbalance/on_unbalanced.rs

```diff
@@ -35,11 +35,26 @@ pub trait OnUnbalanced<Imbalance: TryDrop> {
 	/// Handler for some imbalances. The different imbalances might have different origins or
 	/// meanings, dependent on the context. Will default to simply calling on_unbalanced for all
 	/// of them. Infallible.
-	fn on_unbalanceds<B>(amounts: impl Iterator<Item = Imbalance>)
+	fn on_unbalanceds(mut amounts: impl Iterator<Item = Imbalance>)
 	where
-		Imbalance: crate::traits::Imbalance<B>,
+		Imbalance: crate::traits::tokens::imbalance::TryMerge,
 	{
-		Self::on_unbalanced(amounts.fold(Imbalance::zero(), |i, x| x.merge(i)))
+		let mut sum: Option<Imbalance> = None;
+		while let Some(next) = amounts.next() {
+			sum = match sum {
+				Some(sum) => match sum.try_merge(next) {
+					Ok(sum) => Some(sum),
+					Err((sum, next)) => {
+						Self::on_unbalanced(next);
+						Some(sum)
+					},
+				},
+				None => Some(next),
+			}
+		}
+		if let Some(sum) = sum {
+			Self::on_unbalanced(sum)
```

Comment:
Where is the advantage of calling `on_unbalanced` on the merged result vs calling it on every balance individually?

Reply:
We were merging `tip` and `fee`, to handle both with a single `on_unbalanceds` handler. This call would result a single `deposit` instead of two if the imbalances are identical.

Current open PR with impl of payment handler for the fees, requires two on unbalanced impl, because `on_unbalanceds` cannot handle `fungibles` iterator - https://github.com/paritytech/polkadot-sdk/blob/3a8e675e9f6f283514c00c14d3d1d90ed5bf59c0/substrate/frame/transaction-payment/asset-conversion-tx-payment/src/payment.rs#L267 
pr https://github.com/paritytech/polkadot-sdk/pull/4488

I do not like the breaking change, but also feel like this change needed, since we moving away from `currency` impls and using `fungible` and `fungibles`.

I also wanna have this decided, before I merge that PR to not break it again later.

Reply:
Okay seems fine to me.